ui/contexts/metamask-notifications/metamask-notifications.tsx

```diff
@@ -31,48 +32,85 @@ export const useMetamaskNotificationsContext = () => {
   return context;
 };
 
-export const MetamaskNotificationsProvider: React.FC = ({ children }) => {
+export function useBasicFunctionalityDisableEffect() {
+  const isBasicFunctionalityEnabled = useSelector(getUseExternalServices);
   const isNotificationsEnabled = useSelector(
     selectIsMetamaskNotificationsEnabled,
   );
-  const isBasicFunctionalityEnabled = useSelector(getUseExternalServices);
-  const isUnlocked = useSelector(getIsUnlocked);
-  const isSignedIn = useSelector(selectIsSignedIn);
-
-  const { listNotifications, notificationsData, isLoading, error } =
-    useListNotifications();
   const { disableNotifications } = useDisableNotifications();
+  const { listNotifications } = useListNotifications();
 
-  // Basic functionality effect
   useEffect(() => {
-    if (!isBasicFunctionalityEnabled && isNotificationsEnabled) {
-      // Disable notifications
-      disableNotifications();
-      // list notifications to reset the counter
-      listNotifications();
-    }
+    const run = async () => {
+      try {
+        if (!isBasicFunctionalityEnabled && isNotificationsEnabled) {
+          // disable notifications services
+          await disableNotifications();
+          // list notifications to reset the counter
+          await listNotifications();
+        }
+      } catch {
+        // Do nothing
+      }
+    };
+    run();
   }, [
+    disableNotifications,
     isBasicFunctionalityEnabled,
     isNotificationsEnabled,
-    disableNotifications,
     listNotifications,
   ]);
+}
 
-  const shouldFetchNotifications = useMemo(
-    () => isNotificationsEnabled && isSignedIn,
-    [isNotificationsEnabled, isSignedIn],
+export function useFetchInitialNotificationsEffect() {
+  const isNotificationsEnabled = useSelector(
+    selectIsMetamaskNotificationsEnabled,
   );
+  const isBasicFunctionalityEnabled = useSelector(getUseExternalServices);
+  const isUnlocked = useSelector(getIsUnlocked);
+  const isSignedIn = useSelector(selectIsSignedIn);
+  const shouldFetchNotifications =
+    Boolean(isNotificationsEnabled) && Boolean(isSignedIn);
+  const { enableNotifications } = useEnableNotifications();
+  const { listNotifications } = useListNotifications();
 
   useEffect(() => {
-    if (isBasicFunctionalityEnabled && shouldFetchNotifications && isUnlocked) {
-      listNotifications();
-    }
+    const run = async () => {
+      try {
+        if (
+          isBasicFunctionalityEnabled &&
+          shouldFetchNotifications &&
+          isUnlocked
+        ) {
+          // Re-enabling notifications as we need to ensure that the notification subscriptions are correctly setup
+          // NOTE - we should move this inside the controller on startup
+          await enableNotifications();
```

Comment:
As discussed, I think it would be good to move this into the controller.
However the controller would need some work to support this.

Reply:
Hmmm thinking about this more, it kinda is nice that this is fired in the UI.
The background will be a bit weird to work in (we don't want notification APIs to fire during onboarding), and there are discrepencies between extension and mobile on this.

Maybe we can keep this in the UI but have a resubscription expiry.

Reply:
Resolving - opted to use the effect + expiry approach.